src/main/host/thread_preload.c

```diff
@@ -113,10 +113,24 @@ void threadpreload_free(Thread* base) {
     worker_countObject(OBJECT_TYPE_THREAD_PRELOAD, COUNTER_TYPE_FREE);
 }
 
+static gchar** _add_shadow_pid_to_env(gchar **envp) {
+
+    enum { BUF_NBYTES = 256 };
+    char strbuf[BUF_NBYTES] = {0};
+
+    pid_t shadow_pid = getpid();
+
+    snprintf(strbuf, BUF_NBYTES, "%llu", (unsigned long long)shadow_pid);
+
+    envp = g_environ_setenv(envp, "SHADOW_PID", strbuf, TRUE);
+
+    return envp;
+}
+
 static pid_t _threadpreload_fork_exec(ThreadPreload* thread, const char* file, char* const argv[],
                                       char* const envp[]) {
     pid_t shadow_pid = getpid();
-    pid_t pid = fork();
+    pid_t pid = vfork();
```

Comment:
Maybe add a short comment about why we are using `vfork` since moving away from it isn't a decision that should be taken lightly.

Reply:
Fixed